src/test/resources/com/puppycrawl/tools/checkstyle/grammar/java19/ExpectedRecordPatternsPreviewNestedDecomposition.txt

```diff
@@ -240,9 +292,71 @@ COMPILATION_UNIT -> COMPILATION_UNIT [2:0]
         |       |   |   `--LAND -> && [49:34]
         |       |   |       |--LAND -> && [48:16]
         |       |   |       |   |--LITERAL_INSTANCEOF -> instanceof [46:14]
-        |       |   |       |   |   `--IDENT -> r [46:12]
+        |       |   |       |   |   |--IDENT -> r [46:12]
+        |       |   |       |   |   `--RECORD_PATTERN_DEF -> RECORD_PATTERN_DEF [46:25]
+        |       |   |       |   |       |--MODIFIERS -> MODIFIERS [46:25]
+        |       |   |       |   |       |--TYPE -> TYPE [46:25]
+        |       |   |       |   |       |   `--IDENT -> Rectangle [46:25]
+        |       |   |       |   |       |--LPAREN -> ( [46:34]
+        |       |   |       |   |       |--RECORD_PATTERN_COMPONENTS -> RECORD_PATTERN_COMPONENTS [46:35]
+        |       |   |       |   |       |   |--RECORD_PATTERN_DEF -> RECORD_PATTERN_DEF [46:35]
+        |       |   |       |   |       |   |   |--MODIFIERS -> MODIFIERS [46:35]
+        |       |   |       |   |       |   |   |--TYPE -> TYPE [46:35]
+        |       |   |       |   |       |   |   |   `--IDENT -> ColoredPoint [46:35]
+        |       |   |       |   |       |   |   |--LPAREN -> ( [46:47]
+        |       |   |       |   |       |   |   |--RECORD_PATTERN_COMPONENTS -> RECORD_PATTERN_COMPONENTS [46:48]
```

Comment:
@rnveach example of multiple components

Reply:
Thanks I figured I was missing it. That was my only real concern. I feel fine with what you are doing.